catalog/ui/catalog-ui-filter/src/main/java/org/codice/ddf/catalog/ui/filter/impl/visit/VisitableXmlElementImpl.java

```diff
@@ -452,16 +455,43 @@ private LiteralElement(JAXBElement element) {
             INVALID_INVOCATION + element.getDeclaredType().getName());
       }
       this.value =
-          Optional.of(element)
-              .map(JAXBElement::getValue)
-              .map(LiteralType.class::cast)
-              .map(LiteralType::getContent)
-              .orElseThrow(NullPointerException::new); // Should never occur - see super ctor
+          ((LiteralType) element.getValue())
+              .getContent()
+              .stream()
+              .map(LiteralElement::parsePrimitive)
+              .collect(Collectors.toList());
     }
 
     @Override
     public List<Serializable> getValue() {
       return value;
     }
+
+    private static Serializable parsePrimitive(Serializable literal) {
+      if (literal instanceof String) {
+        return parsePrimitive((String) literal);
+      }
+      return literal;
+    }
+
+    private static Serializable parsePrimitive(String raw) {
+      for (Function<String, Serializable> converter : CONVERTERS) {
+        Serializable result = converter.apply(raw);
+        if (result != null) {
+          return result;
+        }
+      }
+      return raw;
+    }
+
+    private static Serializable tryParseBoolean(String raw) {
+      if ("true".equalsIgnoreCase(raw)) {
```

Comment:
❓ Do you want to accept other values, such as 't', 'f', 'yes', 'no', 'on', 'off'? Or are the strings 'true' and 'false' the only correct inputs? If the former, I'd suggest [`BooleanUtils.toBoolean()`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/BooleanUtils.html#toBoolean-java.lang.String-)